File: src/main/java/scout24/techChallenge/model/Site.java
/**
 *
 */
package scout24.techChallenge.model;

import lombok.Data;

import java.util.HashMap;
import java.util.Map;

/**
 * @author luca
 */
@Data
public class Site {
    private String version;
    private String head;
    private String body;
    private String title;
    private Map<String, Integer> headingTags = new HashMap<>();
    private boolean hasLoginForm = false;
    private int internalLinks = 0;
    private int externalLinks = 0;
    private Map<String, Integer> hyperLinksMap = new HashMap<>();
}

File: web/src/model/Site.ts
export default class Site {
    constructor(
        public version ?: string,
        public head ?: string,
        public body ?: string,
        public title ?: string,
        public headingTags ?: Map<string, string>,
        public hasLoginForm ?: boolean,
        public internalLinks ?: number,
        public externalLinks ?: number,
        public hyperLinksMap ?: Map<string, string>
    ) {
    }
}